callback-heaven.lisp

```diff
@@ -257,29 +257,37 @@ Note that this memory is not further managed!"
             (write-char #\; stream)
             (terpri stream)))
 
-(defun emit-api-function-definitions (ctrans stream)
+(defvar *api-function-body-prefix*  nil)
+(defvar *api-function-body-postfix* nil)
+
+(defun emit-api-function-definitions (ctrans stream &key
+                                        (prefix  *api-function-body-prefix*)
+                                        (postfix *api-function-body-postfix*))
   (let ((index-translations (c-space-translation-index-translations ctrans))
         (api-group (c-space-translation-api-group ctrans)))
     (loop :for i :from 0
           :for fname :across index-translations
           :for f := (api-group-function api-group fname)
           :do (terpri stream)
               (emit-api-function-prototype f stream)
-              (write-string " {" stream)
-              (terpri stream)
-              (format stream "~4T")
-              (unless (eq :void (api-function-return-type f))
-                (write-string "return " stream))
-              (format stream "((~A)(~A[~D]))(~{~A~^, ~});"
+              (format stream " {~%")
+              (format stream "    ~A ret;~%" (type-name-to-foreign (api-function-return-type f)))
+              (when prefix
+                (if (functionp prefix)
+                    (funcall prefix stream ctrans f)
```

Comment:
maybe a little awkward if the `ctrans` and `f` types / accessors aren't exported